src/yup.test.ts

```diff
@@ -72,20 +81,101 @@ describe('yupResolver', () => {
       values: {
         name: 'jimmy',
         age: 24,
+        password: '[}tehk6Uor',
         foo: [{ yup: true }],
         createdOn: new Date('2014-09-23T19:25:25Z'),
       },
     });
   });
 
-  it('should get errors', async () => {
-    const data = {
-      name: 2,
-      age: 'test',
-      createdOn: null,
-      foo: [{ loose: null }],
-    };
-    expect(await yupResolver(schema)(data)).toMatchSnapshot();
+  describe('errors', () => {
+    it('should get errors with validate all criteria fields', async () => {
+      const data = {
+        name: 2,
+        age: 'test',
+        password: '',
+        createdOn: null,
+        foo: [{ loose: null }],
+      };
+      const resolve = await yupResolver(schema)(data, {}, true);
+      expect(resolve).toMatchSnapshot();
+      expect(resolve.errors['age'].types).toMatchInlineSnapshot(`
+          Object {
+            "typeError": Array [
+              "age must be a \`number\` type, but the final value was: \`NaN\` (cast from the value \`\\"test\\"\`).",
+            ],
+          }
+        `);
+      expect(resolve.errors['createdOn'].types).toMatchInlineSnapshot(`
+        Object {
+          "typeError": Array [
+            "createdOn must be a \`date\` type, but the final value was: \`Invalid Date\`.",
+          ],
+        }
+      `);
+      expect(resolve.errors['password'].types).toMatchInlineSnapshot(`
+          Object {
+            "matches": Array [
+              "Special",
+              "Number",
+              "Uppercase",
+              "Lowercase",
+            ],
+            "min": Array [
+              "password must be at least 8 characters",
+            ],
+            "required": Array [
+              "password is a required field",
+            ],
+          }
+        `);
+    });
```

Comment:
The structure I had in mind was:

```ts
expect(resolve.errors['password'].types).toMatchInlineSnapshot(`
  Object {
    "matches": Array [
      "Special",
      "Number",
      "Uppercase",
      "Lowercase",
    ],
    "min": "password must be at least 8 characters",
    "required": "password is a required field",
  }
`);
```

We should discuss whether all should be array message when `validateAllFieldCriteria` is true.

Reply:
Thanks, @kotarella1110. Above is a breaking change, we should stick with 

```
expect(resolve.errors['password'].types).toMatchInlineSnapshot(`
  Object {
    "matches": Array [
      "Special",
      "Number",
      "Uppercase",
      "Lowercase",
    ],
    "min": "password must be at least 8 characters",
    "required": "password is a required field",
  }
`);
```

Reply:
@bluebill1049 I agree with you 👍

Reply:
Yes, i was thinking if we keep this mix structure, i'm afraid that we have to handle the case where `messages` can be an array or a string in the code for every property in the schema. This is why i keep every messages in array format when `validateAllCriteria` is set to true. But you are in better place to take the right decision here :).

Reply:
Thanks very much for understanding and putting this PR together!